images/bootstrap/entrypoint.sh

```diff
@@ -17,8 +17,24 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-# get test-infra for latest bootstrap etc
-git clone https://github.com/kubernetes/test-infra
+BOOTSTRAP_FETCH_TEST_INFRA=${BOOTSTRAP_FETCH_TEST_INFRA:-"true"}
+
+# clone-or-fetch latest test-infra repo
+if ! [[ -d test-infra/.git ]] || [[ "${BOOTSTRAP_FETCH_TEST_INFRA}" != "true" ]]; then
+  echo "cloning https://github.com/kubernetes/test-infra"
+  rm -rf test-infra
+  git clone https://github.com/kubernetes/test-infra
```

Comment:
@spiffxp it seems unlikely we would hit this case, but in the event that we want latest and local repo does not exist (so we must clone it), can we just do a shallow clone?

Reply:
This is here for fallback to previous behavior in the event we see something weird.  Given that bootstrap is deprecated I'd rather not fallback to different clever behavior.

We could consider a followup where a shallow clone is done by default instead of a fetch.  I'm slightly wary of shallow clones since at our scale they can put a hurt on github if there is followup fetching (this is really more of a kubernetes/kubernetes problem).

We could also consider a plan where there is _no_ fetching done at runtime.  What's in the image is what you get.  This is more in line with how `kubetest` changes work today.

It would also be helpful to get stats out of this.  Right now bootstrap seems to be swallowing my attempts to prepend to `build-log.txt`.  Another idea might be to pre-create whatever artifacts dir it uses and dump something in there.

But I consider these all followup.  I'd like to get this included in the kubekins that gets included in today's autobump

Reply:
SGTM